cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -66,7 +64,13 @@ def __init__(self, dual_var, neural_net_param_object, test_input, true_class,
       self.lower.append(current_lower)
       self.upper.append(current_upper)
 
+
+
     # Computing the optimization terms
+    type_check(dual_var['lambda_pos'])
```

Comment:
For my own codebase reverse engineering purposes, I verified that these are all Variables